pkg/resources/object_renaming_lists_and_sets.go

```diff
@@ -0,0 +1,518 @@
+		var name string
+		if nameValue, ok := item.(map[string]any)["name"]; ok {
+			name = nameValue.(string)
+		}
+		return objectRenamingDatabaseListItem{
+			Name:   name,
+			String: item.(map[string]any)["string"].(string),
+			Int:    item.(map[string]any)["int"].(int),
+		}
+	})
+
+	ObjectRenamingDatabaseInstance.OrderedList = objectRenamingDatabaseOrderedListFromSchema(d.Get("ordered_list").([]any))
+	ObjectRenamingDatabaseInstance.ManuallyOrderedList = objectRenamingDatabaseManuallyOrderedListFromSchema(d.Get("manually_ordered_list").([]any))
+
+	d.SetId("identifier")
+
+	return ReadObjectRenamingListsAndSets(ctx, d, meta)
+}
+
+func UpdateObjectRenamingListsAndSets(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	if d.HasChange("list") {
+		// It wasn't working with d.getChange(). It was returning null elements in one of the test case's steps.
+		oldList := d.GetRawState().AsValueMap()["list"].AsValueSlice()
+		newList := d.GetRawConfig().AsValueMap()["list"].AsValueSlice()
+
+		oldListMapped := mapObjectRenamingDatabaseListItemFromValue(oldList)
+		newListMapped := mapObjectRenamingDatabaseListItemFromValue(newList)
+
+		addedItems, removedItems := ListDiff(oldListMapped, newListMapped)
+
+		for _, removedItem := range removedItems {
+			ObjectRenamingDatabaseInstance.List = slices.DeleteFunc(ObjectRenamingDatabaseInstance.List, func(item objectRenamingDatabaseListItem) bool {
+				return item == removedItem
+			})
+		}
+
+		ObjectRenamingDatabaseInstance.List = append(ObjectRenamingDatabaseInstance.List, addedItems...)
+	}
+
+	if d.HasChange("ordered_list") {
+		oldOrderedList := d.GetRawState().AsValueMap()["ordered_list"].AsValueSlice()
+		newOrderedList := d.GetRawConfig().AsValueMap()["ordered_list"].AsValueSlice()
+		oldOrderedListMapped := mapObjectRenamingDatabaseOrderedListItemFromValue(oldOrderedList)
+		newOrderedListMapped := mapObjectRenamingDatabaseOrderedListItemFromValue(newOrderedList)
+
+		itemsToAdd, itemsToRemove := ListDiff(oldOrderedListMapped, newOrderedListMapped)
+		for _, removedItem := range itemsToRemove {
+			ObjectRenamingDatabaseInstance.OrderedList = slices.DeleteFunc(ObjectRenamingDatabaseInstance.OrderedList, func(item objectRenamingDatabaseOrderedListItem) bool {
+				return item == removedItem
+			})
+		}
+
+		for _, addedItem := range itemsToAdd {
+			addedItem.Order = ""
+			ObjectRenamingDatabaseInstance.OrderedList = append(ObjectRenamingDatabaseInstance.OrderedList, addedItem)
+		}
+
+		// The implementation is not complete due to mentioned issues with computed order in DiffSuppressFunc
+	}
+
+	if d.HasChange("manually_ordered_list") {
+		_, newManuallyOrderedList := d.GetChange("manually_ordered_list")
+		hasNewItems := len(newManuallyOrderedList.([]any)) > len(ObjectRenamingDatabaseInstance.ManuallyOrderedList)
+
+		// Copy the value from external source and compare with the new state to create the final state that will be saved
```

Comment:
question: this is nice for testing but we may not have the order from Snowflake, right? How this would behave if the external source does not provide the order?